lalrpop/src/build/mod.rs

```diff
@@ -87,6 +87,36 @@ fn gen_resolve_file(session: &Session, lalrpop_file: &Path, ext: &str) -> io::Re
         in_dir
     };
 
+    // Ideally we do something like syn::parse_str::<syn::Ident>(lalrpop_file.file_name())?;
+    // But I don't think we want a full blown syn dependency unless fully converting to proc macros.
+    if lalrpop_file
+        .file_name()
+        .ok_or(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            format!(
+                "LALRPOP could not extract a valid file name: {}",
+                lalrpop_file.display()
+            ),
+        ))?
+        .to_str()
+        .ok_or(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            format!(
+                "LALRPOP file names must be valid UTF-8: {}",
+                lalrpop_file.display()
+            ),
+        ))?
+        .contains(char::is_whitespace)
+    {
```

Comment:
```suggestion
    let file_name = if let Ok(lalrpop_file) else {
        return Err(io::Error::new(
            io::ErrorKind::InvalidInput,
            format!(
                "LALRPOP could not extract a valid file name: {}",
                lalrpop_file.display()
            )
        );
    };
    let file_name = if let Ok(file_name) = file_name.to_str() else {
        return Err(io::Error::new(
            io::ErrorKind::InvalidInput,
            format!(
                "LALRPOP file names must be valid UTF-8: {}",
                lalrpop_file.display()
            ),
        ));
    };
    if file_name.contains(char::is_whitespace) {
```
how about this style to separate each validation steps?

Reply:
Just my two cents, but isn't the `ok_or()` style more idiomatic rust?  I personally find the version @Pat-Lafon has written to be more readable...